### PR TITLE
feat: add git package

### PR DESCRIPTION
**Pull request by verbanicm**

### CODE REVIEW

File: pkg/git/git.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+// NewGitClient creates a new Terraform client.
+func NewGitClient() *GitClient {
+	return &GitClient{}
+}
+
+// DiffDirs runs a git diff between two revisions and returns the sorted ist of directories with changes.
+func (g *GitClient) DiffDirs(ctx context.Context, workingDir, baseRef, headRef string) ([]string, error) {
+	matches := make(map[string]struct{})
+
+	result, err := child.Run(ctx, &child.RunConfig{
```
Comment (verbanicm) [inline pkg/git/git.go:51]: I think adding an interface for child might make sense to we can mock commands to provide better test coverage in the future.

Thoughts?

File: pkg/git/git.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		if len(line) > 0 {
+			matches[filepath.Dir(line)] = struct{}{}
+		}
+	}
+
+	dirs := maps.Keys(matches)
+
+	sort.Strings(dirs)
+
+	return dirs, nil
```
Comment (yolocs) [inline pkg/git/git.go:75]: Would recommend making a helper func of it and add basic unit test.

File: pkg/git/git.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+// GitClient implements the git interface.
+type GitClient struct{}
+
+// NewGitClient creates a new Terraform client.
+func NewGitClient() *GitClient {
+	return &GitClient{}
+}
+
+// DiffDirs runs a git diff between two revisions and returns the sorted ist of directories with changes.
+func (g *GitClient) DiffDirs(ctx context.Context, workingDir, baseRef, headRef string) ([]string, error) {
```
Comment (sethvargo) [inline pkg/git/git.go:48]: I'm having nightmares about the order of arguments for `ln`... 3 string properties in a row feels like a recipe for misuse. Should we accept a struct instead?

Comment (verbanicm) [reply pkg/git/git.go:48]: I can move the working dir to be a property of the struct? Then its just 2 strings?

File: pkg/git/git.go (hunk 2 above)
Comment (verbanicm) [reply pkg/git/git.go:75]: SGTM, this was my fallback instead of an interface. Ill update.

File: pkg/git/git.go (hunk 3 above)
Comment (verbanicm) [reply pkg/git/git.go]: moved

File: pkg/git/git_test.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	t.Parallel()
+
+	cases := []struct {
+		name  string
+		value string
+		exp   []string
+	}{
+		{
+			name:  "success",
+			value: "first/test.txt\nsecond/test.txt\nthird/test.txt",
```
Comment (yolocs) [inline pkg/git/git_test.go:33]: nit: you could do
```suggestion
			value: `first/test.txt
second/test.txt
third/test.txt`,
```

**Review by yolocs — DISMISSED** [review]

**Review by sethvargo — APPROVED** [review]

**Pull request merged**